src/serializer/ResidualFunctions.js

```diff
@@ -117,6 +118,52 @@ export class ResidualFunctions {
     if (!this.firstFunctionUsages.has(val)) this.firstFunctionUsages.set(val, bodyReference);
   }
 
+  _shouldUseFactoryFunction(funcBody: BabelNodeBlockStatement, instances: Array<FunctionInstance>) {
+    function shouldInlineFunction(): boolean {
+      let shouldInline = true;
+      if (funcBody.start && funcBody.end) {
+        let bodySize = funcBody.end - funcBody.start;
+        shouldInline = bodySize <= 30;
+      }
+      return shouldInline;
+    }
+    let functionInfo = this.residualFunctionInfos.get(funcBody);
+    invariant(functionInfo);
+    let { usesArguments } = functionInfo;
+    return !shouldInlineFunction() && instances.length > 1 && !usesArguments;
+  }
+
+  // Note: this function takes linear time. Please do not call it inside loop.
+  _hasRewrittenFunctionInstance(
```

Comment:
You should be able to change this to `instances.length === 1 && rewrittenAdditionalFunctions.has(instances[0].functionValue)` because each additional FunctionValue should correspond to exactly one FunctionInstance and we don't know before spliceFunctions what the full body of an additional function is so I assume they're all different.